### PR TITLE
87724 referral intro mock integration

### PR DESCRIPTION
**Pull request by lee-delarm6**
# Summary

- Adding mock data
- Adding service level call
- Filling in particular page related to this (referral review)

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/87724

## Testing done

- Added mocks
- Visual testing
- Empty set testing

## Screenshots

Screenshot 2024-07-16 at 2 34 27 PM
Screenshot 2024-07-17 at 10 14 57 AM
Screenshot 2024-07-17 at 10 14 43 AM

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Empty states are handled properly

### CODE REVIEW

File: src/applications/vaos/services/referral/index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { apiRequestWithUrl } from '../utils';
+
+const acheronHeader = {
```
Comment (simiadebowale) [inline src/applications/vaos/services/referral/index.js:3]: We don't need Acheron headers for calls to wellhive. These are only needed for calls to vistA

File: src/applications/vaos/services/referral/index.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  headers: { ACHERON_REQUESTS: 'true' },
+};
+
+export async function getPatientDetails(
```
Comment (simiadebowale) [inline src/applications/vaos/services/referral/index.js:7]: I think what we really need for this PR is to fetch a single referral by id. We can rename this to getReferral (to fetch a single referral by id).  
My guess is we should only need referralID as a param, the ID will be gotten from the previous page.

File: src/applications/vaos/services/referral/index.js (hunk 1 above)
Comment (lee-delarm6) [reply src/applications/vaos/services/referral/index.js:3]: Gotcha!

File: src/applications/vaos/services/mocks/index.js [hunk 3]
```diff
@@ -451,6 +452,16 @@ const responses = {
   },
 
   // WellHive api
+  'GET /vaos/v2/wellhive/referralDetails': (req, res) => {
+    return res.json({
+      data: patientDetails.data,
+    });
+  },
+  'GET /vaos/v2/wellhive/referralDetails/:referralId': (req, res) => {
+    return res.json({
```
Comment (simiadebowale) [inline src/applications/vaos/services/mocks/index.js:461]:
```suggestion
    return res.json({
    data: patientDetails.data.referrals.find(
        referral => referral?.id === req.params.referralId,
      ),
    });```

File: src/applications/vaos/services/mocks/wellHive/patientBasicReferralDetails.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "data": {
+    "referral": {
```
Comment (simiadebowale) [inline src/applications/vaos/services/mocks/wellHive/patientBasicReferralDetails.json:3]: we can add details as part of the referrals array and add multiple referrals assuming api returns multiple referrals
```suggestion
    "referrals": [
      {
      "id": "mock123",
      "typeOfCare": "Primary Care",
      "referralNumber": 123456,
      "startDate": "2021-01-01",
      "expirationDate": "2021-02-01",
      "referringFacility": "White River VA Clinic, VT",
      "phoneNumber": "555-555-5555",
      "detailsShared": {
        "preferredProvider": "Dr. Smith",
        "preferredLocation": "123 Main St, Springfield, IL 62704",
        "preferredFacility": "Springfield VA Clinic",
        "bookNotes": "Patient has a history of back pain"
      }
    },
    {
      "id": "mock456",
      "typeOfCare": "Audiology",
      "referralNumber": 67890,
      "startDate": "2024-11-01",
      "expirationDate": "2024-12-01",
      "referringFacility": "White River VA Clinic, VT",
      "phoneNumber": "234-567-8900",
      "detailsShared": {
        "preferredProvider": "Dr. Smith",
        "preferredLocation": "123 Main St, Springfield, IL 62704",
        "preferredFacility": "Springfield VA Clinic",
        "bookNotes": "Patient has a history of back pain"
      }
    }
  ]
```

File: src/applications/vaos/services/mocks/index.js [hunk 5]
```diff
@@ -39,6 +39,7 @@ const confirmedV2 = require('./v2/confirmed.json');
 
 // CC Direct Scheduling mocks
 const wellHiveAppointments = require('./wellHive/appointments.json');
+const patientDetails = require('./wellHive/patientBasicReferralDetails.json');
```
Comment (simiadebowale) [inline src/applications/vaos/services/mocks/index.js:42]: Do you think we should rename this variable & json file to referralDetails or something that more accurately describes the data it returns

**Review by simiadebowale — COMMENTED** [review]
Added a few comments

File: src/applications/vaos/services/mocks/index.js (hunk 5 above)
Comment (lee-delarm6) [reply src/applications/vaos/services/mocks/index.js]: I changed this up too ^

**Review by simiadebowale — APPROVED** [review]
LGTM!

**Pull request merged**